src/components/reviewMatches/SearchModal.tsx

```diff
@@ -0,0 +1,82 @@
+import { Box, Button, Divider, Grid, Modal, Typography } from '@mui/material'
+import CustomSearchForm from 'components/customSearch/CustomSearchForm'
+import SimpleSearchForm from 'components/search/SimpleSearchForm'
+import { FC, useCallback, useState } from 'react'
+import {
+  CustomSearchQuery,
+  SearchQuery,
+  ToggleOptionsProps
+} from 'types/SimpleSearch'
+import SearchTypeToggle from './SearchTypeToggle'
+
+const style = {
+  position: 'absolute',
+  minWidth: '900px',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24
+}
+
+const options: ToggleOptionsProps[] = [
+  { value: 0, label: 'Custom search' },
+  { value: 1, label: 'Simple search' }
+]
+
+const SearchModal: FC<{ isOpen: boolean; onClose: () => void }> = ({
+  isOpen,
+  onClose
+}) => {
+  const [tabValue, setTabValue] = useState('custom-search')
+  const [refineSearchQuerry, setRefineSearchQuerry] = useState<
+    SearchQuery | CustomSearchQuery | undefined
+  >(undefined)
+
+  const onSearchQuerryChange = useCallback(
+    (value: SearchQuery | CustomSearchQuery) => {
+      setRefineSearchQuerry(value)
+    },
+    []
+  )
```

Comment:
I wonder why we use useCallback here, cause it may lead to errors. This call will keep reference to the first "setRefineSearchQuerry"